eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/VariantExporterController.java

```diff
@@ -67,21 +94,142 @@ public QueryOptions getQuery(MultivaluedMap<String, String> queryParameters) {
+            exportChromosomeVariants(writers, chromosome);
+        }
+
+        writers.values().forEach(VariantContextWriter::close);
+    }
+
+    private void exportChromosomeVariants(Map<String, VariantContextWriter> writers, String chromosome) {
+        List<Region> allRegionsInChromosome = regionFactory.getRegionsForChromosome(chromosome);
+        for (Region region : allRegionsInChromosome) {
+            VariantDBIterator regionVariantsIterator = variantDBAdaptor.iterator(getRegionQuery(region));
+            Map<String, List<VariantContext>> exportedVariants = exporter.export(regionVariantsIterator, region, studies);
+            failedVariants += exporter.getFailedVariants();
+            writeRegionVariants(writers, exportedVariants);
+        }
+    }
+
+    private void writeRegionVariants(Map<String, VariantContextWriter> writers, Map<String, List<VariantContext>> exportedVariants) {
+        exportedVariants.forEach((study, variants) -> {
+            if (writers.containsKey(study)) {
+                // TODO: test if is better if we sort into a new collection instead of use the same
+                Collections.sort(variants, (v1, v2)-> v1.getStart() - v2.getStart());
+                VariantContextWriter writer = writers.get(study);
+                variants.forEach(writer::add);
+            }
+        });
+    }
+
+    private QueryOptions getRegionQuery(Region region) {
+        QueryOptions regionQuery = new QueryOptions(query);
+        regionQuery.put(VariantDBAdaptor.REGION, region.toString());
+        return regionQuery;
+    }
+
+    private void checkHeaders(Map<String, VCFHeader> headers) {
+        // TODO: use this exception or something different?
+        for (String studyId : studies) {
+            if (!headers.containsKey(studyId)) {
+                throw new IllegalArgumentException("Aborting VCF export: missing header for study " + studyId);
+            }
+        }
+    }
+
+    private Map<String, VariantContextWriter> getWritters(Map<String, VCFHeader> headers) {
+        String suffix = ".exported.vcf.gz";
+        Map<String, VariantContextWriter> writers = new TreeMap<>();
+        outputFilePaths = new HashMap<>();
+
+        // setup writers
+        for (String studyId : studies) {
+
+            Path outputPath = Paths.get(outputDir).resolve(studyId + suffix);
+            outputFilePaths.put(studyId, outputPath.toString());
+            // TODO: does the get sequence dictionary method throw any exception?
+//            // get sequence dictionary from header
+//            SAMSequenceDictionary sequenceDictionary;
+//            try {
+//                sequenceDictionary = headers.get(studyId).getSequenceDictionary();
+//            } catch (Exception e) {
+//                // TODO: log error
+//                sequenceDictionary = null;
+//            }
+            SAMSequenceDictionary sequenceDictionary = headers.get(studyId).getSequenceDictionary();
```

Comment:
Example of contig line (human GRCh37 chromosome Y): 

#contig=<ID=Y,length=59373566,assembly=GCA_000001405.1,accession=CM000686.1>

Like incorrect contig lines are common, and contig lines are not mandatory in the VCF specification, then we should catch the exception to avoid the VCF exporter finishing with errors.